lib/kubernetes-deploy/kubernetes_resource.rb

```diff
@@ -192,5 +193,10 @@ def pretty_status
     def kubectl
       @kubectl ||= Kubectl.new(namespace: @namespace, context: @context, logger: @logger, log_failure_by_default: false)
     end
+
+    def template
```

Comment:
I put it here because multiple subclasses in this PR use it:
- `ReplicaSet` uses it to grab the container names in the case where it was not derived from a deployment (managed ReplicaSets will not have a template file)
- `Service` uses it to find the related deployment so it can be smarter about how many endpoints to expect (notably so that it can expect zero when appropriate)

Your comment make me think that perhaps I should move the fallback right into this method: if we have a file, load it from file; if we don't, fetch it from the API.